docs/proposals/20191030-machine-health-checking.md

```diff
@@ -167,7 +164,7 @@ type target struct {
 - Annotates unhealthy target machines as described above.
 
 Out of band:
-- The owning controller observes the annotation and is responsible to remediate the machine.
+- The owning controller observes the condition and is responsible to remediate the machine.
 - The owning controller performs any pre-deletion tasks required.
```

Comment:
What is expected from the owning controller if the Machine becomes healthy again during/before some of the "pre-deletion" tasks?

What do we expect from MHC to do if it marked the Machine as unhealthy with that condition and later it becomes healthy again?

Reply:
This is the same as with annotations, machines do not become healthy again right now, they just are deleted. If it passes a MHC after it fails, it will still get deleted.

Reply:
I understand that this is the goal but I'm curios about the implementation details.
What is expected from MHC if unhealthy Machine is healthy again? should it change the status of `ConditionHealthCheckSucceeded` from `False` to `True`?

Reply:
I think the intention is that the MHC should never change this condition, once it's been added and set to false, that's it, that Machine is going to go away, there's no point health checking it anymore.

I think once external remediation is implemented, it will need to be reconsidered, but for the purposes of this proposal, having it "irreversible" is sufficient

Reply:
Makes sense. I think this should be explicitly specified in the proposal

Reply:
I actually think we should update it to true. There's always a chance that the controller has not done remediation yet, so in the rare event that it fixes itself and hasn't been remediated yet we should leave it alone

Reply:
What if a controller has only performance partial remediation (prior to the deletion of the Machine resource) at the point that the Machine is returned to healthy? Do we need to worry about the possibility of the Machine being treated as healthy when it may actually now be in a degraded state from the partial remediation?

In a simplistic case, what if we have already cordoned/drained a given Node associated with a Machine but have not yet performed the deletion when the MHC now marks the Machine as ready, if for some reason the cordoning/draining and machine deletion are handled through re-entrancy rather than atomically, we could end up with a situation where we've cordoned/drained the Node, but fail to delete the Machine and now that the Machine is "healthy" we fail to delete the Machine, but also have a Machine that is also not capable of handling workloads due to being cordoned.

Reply:
We can punt this to the external remediation proposal and never change machines back to healthy for now